runtime/Makefile

```diff
@@ -59,13 +59,9 @@ $(CHPL_ENV_HEADER): $(CHPL_MAKE_HOME)/util/printchplenv $(CHPL_MAKE_HOME)/util/c
 	@echo "#ifndef _CHPL_ENV_GEN_H_" >> $(CHPL_ENV_HEADER)
 	@echo "#define _CHPL_ENV_GEN_H_" >> $(CHPL_ENV_HEADER)
 	@$(CHPL_MAKE_HOME)/util/printchplenv --runtime --no-tidy --anonymize --simple | \
-          grep -v CHPL_HOST_CC | \
-          grep -v CHPL_HOST_CXX | \
-          grep -v CHPL_TARGET_CC | \
-          grep -v CHPL_TARGET_CXX | \
-          sed s/[-+.\/]/_/g | \
-          sed s/\=/' '/ | \
-          awk '{ print "#define " $$1 "_" toupper($$2) }' >> $(CHPL_ENV_HEADER)
+	  sed 's/^[ \t]*//;s/[ \t]*$$//' | \
+	  sed 's/[^0-9A-Za-z]/_/g' | \
+	  awk '{ print "#define " toupper($$1) }' >> $(CHPL_ENV_HEADER)
```

Comment:
Could we add a comment describing this? Does it relate to the comment on line 25 of `chpl-env-gen.precomp`? 

> munge characters similarly to Makefile generating them